bdai_ros2_wrappers/bdai_ros2_wrappers/utilities.py

```diff
@@ -450,3 +460,20 @@ def take_kwargs(func: Callable, kwargs: Mapping) -> Tuple[Mapping, Mapping]:
         else:
             dropped[name] = value
     return taken, dropped
+
+
+def ensure(value: Optional[Any]) -> Any:
+    """Ensures `value` is not None or fails trying."""
+    if value is None:
+        frame = inspect.currentframe()
+        assert frame is not None
+        frame = frame.f_back
+        assert frame is not None
+        traceback = inspect.getframeinfo(frame)
+        message = f"{traceback.filename}:{traceback.lineno}: "
+        if traceback.code_context is not None:
+            message += "".join(traceback.code_context).strip("\n ") + " failed"
```

Comment:
Should this traceback block of code be its own function? Seems generic enough.

Reply:
Sure thing. Done in 8ab505fd9114ccda1942d56b2722983f993a5423.